# Descripción del proyecto

Trabajas para la compañía minera OilyGiant. Tu tarea es encontrar el mejor lugar para
un nuevo pozo.
Pasos para elegir la ubicación:
- Recolecta los parámetros del pozo de petróleo en la región seleccionada: calidad
del petróleo y volumen de reservas;
- Construye un modelo para predecir el volumen de reservas en los nuevos pozos;
- Selecciona los pozos de petróleo con los valores estimados más altos;
- Elige la región con el mayor beneficio total para los pozos de petróleo
seleccionados.

Tienes datos sobre muestras de crudo de tres regiones. Ya se conocen los parámetros
de cada pozo petrolero de la región. Crea un modelo que ayude a elegir la región con el
mayor margen de beneficio. Analiza los beneficios y riesgos potenciales utilizando la
técnica bootstrapping.

In [1]:
# Librerías
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Paso 1. Descarga y preparación de los datos

In [2]:
# Leer los datasets
df_01 = pd.read_csv("/datasets/geo_data_0.csv")
df_02 = pd.read_csv("/datasets/geo_data_1.csv")
df_03 = pd.read_csv("/datasets/geo_data_2.csv")


In [3]:
print("Región 1\n")
print()
print(df_01.info())

Región 1


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB
None


In [4]:
print("Región 2\n")
print()
print(df_02.info())

Región 2


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB
None


In [5]:
print("Región 3\n")
print(df_03.info())

Región 3

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB
None


In [6]:
print(df_01.head())

      id        f0        f1        f2     product
0  txEyH  0.705745 -0.497823  1.221170  105.280062
1  2acmU  1.334711 -0.340164  4.365080   73.037750
2  409Wp  1.022732  0.151990  1.419926   85.265647
3  iJLyR -0.032172  0.139033  2.978566  168.620776
4  Xdl7t  1.988431  0.155413  4.751769  154.036647


- No se encontraron valores nulos.
- Los datos estan correctos.

Para propositos de este proyecto, se nombró los dataset de la siguiente manera:
geo_data_0: Región 1,
geo_data_1: Región 2,
geo_data_2: Región 3

# Paso 2. Entrenar y probar el modelo para cada región

## 2.1 Dividir los datos en conjuntos de entrenamiento y validación

### Función

In [7]:
# Función para dividir características y objetivo
def split_features_target(df):
    features = df[['f0', 'f1', 'f2']]
    target = df['product']

    return features, target

In [8]:
def split_train_valid(features, target):
    features_train, features_valid, target_train, target_valid = train_test_split(
        features, 
        target, 
        test_size=0.25, 
        random_state=12345)
    
    return features_train, features_valid, target_train, target_valid

### Región 01

In [9]:
features_reg1, target_reg1 = split_features_target(df_01)
features_train_reg1, features_valid_reg1, target_train_reg1, target_valid_reg1 = split_train_valid(features_reg1, target_reg1)

print(f"Features del conjunto de entrenamiento {features_train_reg1.shape}")
print(f"Features del conjunto de validación {features_valid_reg1.shape}")
print(f"Objetivo del conjunto de entrenamiento {target_train_reg1.shape}")
print(f"Objetivo del conjunto de validación {target_valid_reg1.shape}")

Features del conjunto de entrenamiento (75000, 3)
Features del conjunto de validación (25000, 3)
Objetivo del conjunto de entrenamiento (75000,)
Objetivo del conjunto de validación (25000,)


### Región 02

In [10]:
features_reg2, target_reg2 = split_features_target(df_02)
features_train_reg2, features_valid_reg2, target_train_reg2, target_valid_reg2 = split_train_valid(features_reg2, target_reg2)

print(f"Features del conjunto de entrenamiento {features_train_reg2.shape}")
print(f"Features del conjunto de validación {features_valid_reg2.shape}")
print(f"Objetivo del conjunto de entrenamiento {target_train_reg2.shape}")
print(f"Objetivo del conjunto de validación {target_valid_reg2.shape}")

Features del conjunto de entrenamiento (75000, 3)
Features del conjunto de validación (25000, 3)
Objetivo del conjunto de entrenamiento (75000,)
Objetivo del conjunto de validación (25000,)


### Región 03

In [11]:
features_reg3, target_reg3 = split_features_target(df_03)
features_train_reg3, features_valid_reg3, target_train_reg3, target_valid_reg3 = split_train_valid(features_reg3, target_reg3)

print(f"Features del conjunto de entrenamiento {features_train_reg3.shape}")
print(f"Features del conjunto de validación {features_valid_reg3.shape}")
print(f"Objetivo del conjunto de entrenamiento {target_train_reg3.shape}")
print(f"Objetivo del conjunto de validación {target_valid_reg3.shape}")

Features del conjunto de entrenamiento (75000, 3)
Features del conjunto de validación (25000, 3)
Objetivo del conjunto de entrenamiento (75000,)
Objetivo del conjunto de validación (25000,)


- Se separo las características y los objetivos.
- Finalmente se separo en conjunto de entrenamiento y validación en una propoción de 75:25 respectivamente.
- Esto se realizo para los 3 diferentes datasets.


## 2.2 Entrenar el modelo y predicciones para el conjunto de validación

### Función

In [12]:

# Función para entrenar el modelo
def get_prediction(features_train, target_train, features_valid):
    model = LinearRegression()
    model.fit(features_train, target_train)
    predicted_valid = model.predict(features_valid)

    return predicted_valid


### Región 01

In [13]:
# Obtener las predicciones de la región 1
predicted_valid_reg1 = get_prediction(features_train_reg1, target_train_reg1, features_valid_reg1)

print(predicted_valid_reg1)

[ 95.89495185  77.57258261  77.89263965 ...  61.50983303 118.18039721
 118.16939229]


### Región 02

In [14]:
predicted_valid_reg2 = get_prediction(features_train_reg2, target_train_reg2, features_valid_reg2)

print(predicted_valid_reg2)

[ 82.66331365  54.43178616  29.74875995 ... 137.87934053  83.76196568
  53.95846638]


### Región 03

In [15]:
predicted_valid_reg3 = get_prediction(features_train_reg3, target_train_reg3, features_valid_reg3)

print(predicted_valid_reg3)

[ 93.59963303  75.10515854  90.06680936 ...  99.40728116  77.77991248
 129.03241718]


## 2.3 Guarda las predicciones y las respuestas correctas para el conjunto de validación.

### Región 01

In [16]:
print("Predicciones del conjunto de validación")
print(predicted_valid_reg1)
print()
print("Respuestas del conjunto de validación")
print(target_valid_reg1.head())

Predicciones del conjunto de validación
[ 95.89495185  77.57258261  77.89263965 ...  61.50983303 118.18039721
 118.16939229]

Respuestas del conjunto de validación
71751     10.038645
80493    114.551489
2655     132.603635
53233    169.072125
91141    122.325180
Name: product, dtype: float64


### Región 02

In [17]:
print("Predicciones del conjunto de validación")
print(predicted_valid_reg2)
print()
print("Respuestas del conjunto de validación")
print(target_valid_reg2.head())

Predicciones del conjunto de validación
[ 82.66331365  54.43178616  29.74875995 ... 137.87934053  83.76196568
  53.95846638]

Respuestas del conjunto de validación
71751    80.859783
80493    53.906522
2655     30.132364
53233    53.906522
91141     0.000000
Name: product, dtype: float64


### Región 03

In [18]:
print("Predicciones del conjunto de validación")
print(predicted_valid_reg3)
print()
print("Respuestas del conjunto de validación")
print(target_valid_reg3.head())

Predicciones del conjunto de validación
[ 93.59963303  75.10515854  90.06680936 ...  99.40728116  77.77991248
 129.03241718]

Respuestas del conjunto de validación
71751     61.212375
80493     41.850118
2655      57.776581
53233    100.053761
91141    109.897122
Name: product, dtype: float64


## 2.4 volumen promedio de las reservas previstas y el RECM del modelo.

### Función

In [19]:
# Volumen medio de respuestas predichas
def print_predicted_valid(predicted_valid):
    mean_predicted = predicted_valid.mean()
    print(f"Volumen medio de reservas predicho {mean_predicted:.2f}")

# Volumen medio de respuestas correctas
def print_target_valid(target_valid):
    mean_y_valid = target_valid.mean()
    print(f"Volumne medio de reservas correcto {mean_y_valid:.2f}")

# Imprimir raiz del error cuadratico medio
def print_rmse(target_valid, predicted_valid):
    rmse = mean_squared_error(target_valid, predicted_valid) ** 0.5
    print(f'Raiz del error cuadratico medio {rmse}')

### Región 01

In [20]:
print_predicted_valid(predicted_valid_reg1)
print()
print_target_valid(target_valid_reg1)
print()
print_rmse(target_valid_reg1, predicted_valid_reg1)

Volumen medio de reservas predicho 92.59

Volumne medio de reservas correcto 92.08

Raiz del error cuadratico medio 37.5794217150813


### Región 02

In [21]:
print_predicted_valid(predicted_valid_reg2)
print()
print_target_valid(target_valid_reg2)
print()
print_rmse(target_valid_reg2, predicted_valid_reg2)

Volumen medio de reservas predicho 68.73

Volumne medio de reservas correcto 68.72

Raiz del error cuadratico medio 0.893099286775617


### Región 03

In [22]:
print_predicted_valid(predicted_valid_reg3)
print()
print_target_valid(target_valid_reg3)
print()
print_rmse(target_valid_reg3, predicted_valid_reg3)

Volumen medio de reservas predicho 94.97

Volumne medio de reservas correcto 94.88

Raiz del error cuadratico medio 40.02970873393434


## 2.5 Analiza los resultados

### Región 01

El modelo predice que en promedio cada pozo en la región 01 producirá 92.59 mil barriles de petróleo.
El error promedio del modelo es de 37.57 mil barriles de petróleo. En algunos pozos el modelo puede estar equivocado +- 37.57 unidades.

### Región 02

El modelo predice que en promedio cada pozo en la región 02 producirá 68.73 mil barriles de petróleo.
El error promedio del modelo es de 0.89 mil barriles de petróleo. En algunos pozos el modelo puede estar equivocado +- 0.89 unidades.

### Región 03

El modelo predice que en promedio cada pozo en la región 03 producirá 94.97 mil barriles de petróleo.
El error promedio del modelo es de 40.02 mil barriles de petróleo. En algunos pozos el modelo puede estar equivocado +- 40.02 unidades.

# Paso 3. Cálculo de ganancias

## 3.1 Valores necesarios para los cálculos

In [23]:
total_investment = 100_000_000
price_per_unit = 4500
wells = 200

In [24]:
# Inversión por pozo
investment_per_well = total_investment / wells
print(f"Inversión por pozo: {investment_per_well}")

Inversión por pozo: 500000.0


In [25]:
# Unidades necesarias para recuperar la inversión
units_needed = investment_per_well  / price_per_unit
print(f"Unidades necesarías por pozo {units_needed:.1f}")

Unidades necesarías por pozo 111.1


## 3.2 Comparación entre la cantidad rentable y la cantidad media de cada región

In [26]:
reserves_avg_reg1 = predicted_valid_reg1.mean() 
reserves_avg_reg2 = predicted_valid_reg2.mean() 
reserves_avg_reg3 = predicted_valid_reg3.mean()

print(f"Cantidad media de reservas en la región 1: {reserves_avg_reg1:.2f}")
print(f"Cantidad media de reservas en la región 2: {reserves_avg_reg2:.2f}")
print(f"Cantidad media de reservas en la región 3: {reserves_avg_reg3:.2f}")

Cantidad media de reservas en la región 1: 92.59
Cantidad media de reservas en la región 2: 68.73
Cantidad media de reservas en la región 3: 94.97


In [27]:
# comparativa de la región 1
if reserves_avg_reg1 < units_needed:
    print(f"La Región 01 está {units_needed - reserves_avg_reg1:.1f} unidades por debajo del umbral")
else:
    print("✅ La Región 01 supera el umbral de rentabilidad")

La Región 01 está 18.5 unidades por debajo del umbral


In [28]:
# comparativa de la región 2
if reserves_avg_reg2 < units_needed:
    print(f"La Región 02 está {units_needed - reserves_avg_reg2:.1f} unidades por debajo del umbral")
else:
    print("✅ La Región 02 supera el umbral de rentabilidad")

La Región 02 está 42.4 unidades por debajo del umbral


In [29]:
# comparativa de la región 3
if reserves_avg_reg3 < units_needed:
    print(f"La Región 03 está {units_needed - reserves_avg_reg3:.1f} unidades por debajo del umbral")
else:
    print("✅ La Región 03 supera el umbral de rentabilidad")

La Región 03 está 16.1 unidades por debajo del umbral


## 3.3 Conclusiones sobre cómo preparar el paso para calcular el beneficio:

1. Ninguna región supera el umbral promedio: Las tres regiones están por debajo 
   de las 111.1 unidades necesarias para garantizar rentabilidad por pozo.

2. La Región 03 es la más prometedora: Con 94.97 unidades promedio, es la que 
   está más cerca del umbral de rentabilidad.

3. La Región 02 presenta menor potencial de ganancias: Con solo 68.73 unidades promedio, está muy por debajo del umbral de rentabilidad (111.1 unidades). Aunque tiene el RMSE más bajo (0.89), indicando alta precisión en las predicciones, su baja productividad la hace menos atractiva para la inversión.

# Paso 4. Escribe una función para calcular la ganancia de un conjunto de pozos de petróleo seleccionados y modela las predicciones:

## 4.1 Seleccionar los 200 mejores pozos

### Predicciones

In [30]:
predicted_valid_reg1 = pd.Series(predicted_valid_reg1, index=target_valid_reg1.index)
best2_pred_reg1 = predicted_valid_reg1.sort_values(ascending=False).head(200)
print(best2_pred_reg1.head())

93073    180.180713
46784    176.252213
78948    175.850623
43388    175.658429
6496     173.299686
dtype: float64


In [31]:
predicted_valid_reg2 = pd.Series(predicted_valid_reg2, index=target_valid_reg2.index)
best2_pred_reg2 = predicted_valid_reg2.sort_values(ascending=False).head(200)
print(best2_pred_reg2.head())

38665    139.818970
20191    139.773423
14041    139.703330
24274    139.560938
92782    139.516754
dtype: float64


In [32]:
predicted_valid_reg3 = pd.Series(predicted_valid_reg3, index=target_valid_reg3.index)
best2_pred_reg3 = predicted_valid_reg3.sort_values(ascending=False).head(200)
print(best2_pred_reg3.head())

98619    165.856833
46649    165.679685
82661    163.439962
53151    162.062589
18747    161.797476
dtype: float64


### Real

In [33]:
# 200 pozos de la región 1 reales
best2_wells_real_reg1 = target_valid_reg1.loc[best2_pred_reg1.index]
best2_wells_real_reg2 = target_valid_reg2.loc[best2_pred_reg2.index]
best2_wells_real_reg3 = target_valid_reg3.loc[best2_pred_reg3.index]

print(f"Región 1 - tamaño: {best2_wells_real_reg1.shape[0]}")
print()
print(best2_wells_real_reg1.head())
print("----------------------------------------------------")
print(f"Región 2 - tamaño: {best2_wells_real_reg2.shape[0]}")
print()
print(best2_wells_real_reg1.head())
print("----------------------------------------------------")
print(f"Región 3 - tamaño: {best2_wells_real_reg3.shape[0]}")
print()
print(best2_wells_real_reg3.head())
print("----------------------------------------------------")

Región 1 - tamaño: 200

93073    162.810993
46784    153.639837
78948    162.153488
43388     96.893581
6496     178.879516
Name: product, dtype: float64
----------------------------------------------------
Región 2 - tamaño: 200

93073    162.810993
46784    153.639837
78948    162.153488
43388     96.893581
6496     178.879516
Name: product, dtype: float64
----------------------------------------------------
Región 3 - tamaño: 200

98619    175.103291
46649    131.627481
82661    141.160070
53151    159.676082
18747    142.135203
Name: product, dtype: float64
----------------------------------------------------


## 4.2 Calcular el volumen total de reservas

In [34]:
# Suma de las 200 mejores predicciones de cada región
vol_reg1 = best2_pred_reg1.sum()
vol_reg2 = best2_pred_reg2.sum()
vol_reg3 = best2_pred_reg3.sum()

print(f"Volumen objetivo de la región 1: {vol_reg1:.2f}")
print(f"Volumen objetivo de la región 2: {vol_reg2:.2f}")
print(f"Volumen objetivo de la región 3: {vol_reg3:.2f}")

Volumen objetivo de la región 1: 31102.33
Volumen objetivo de la región 2: 27746.03
Volumen objetivo de la región 3: 29603.90


In [35]:
# Predicciones (200 reservas) por región
print(f"Región 1 - tamaño: {best2_pred_reg1.shape[0]}")
print()
print(best2_pred_reg1.head())
print("------------------------------------------------")
print(f"Región 2 - tamaño: {best2_pred_reg2.shape[0]}")
print()
print(best2_pred_reg2.head())
print("------------------------------------------------")
print(f"Región 3 - tamaño: {best2_pred_reg3.shape[0]}")
print()
print(best2_pred_reg3.head())
print("------------------------------------------------")

Región 1 - tamaño: 200

93073    180.180713
46784    176.252213
78948    175.850623
43388    175.658429
6496     173.299686
dtype: float64
------------------------------------------------
Región 2 - tamaño: 200

38665    139.818970
20191    139.773423
14041    139.703330
24274    139.560938
92782    139.516754
dtype: float64
------------------------------------------------
Región 3 - tamaño: 200

98619    165.856833
46649    165.679685
82661    163.439962
53151    162.062589
18747    161.797476
dtype: float64
------------------------------------------------


## 4.3 Ganancia potencial de los 200 pozos principales por región

In [36]:
# Función para calcular las ganacias
def revenue(units):
    return (units * 4500) - 100_000_000

In [37]:
# Calcular ganancias de la región 1 con los datos reales
revenue_reg1 = revenue(best2_wells_real_reg1.sum())
print(f"Unidades (mil barriles): {best2_wells_real_reg1.sum():.2f}")
print(f"Ganancias: {revenue_reg1:.2f}")


Unidades (mil barriles): 29601.84
Ganancias: 33208260.43


In [38]:
# Calcular ganancias de la región 2 con los datos reales
revenue_reg2 = revenue(best2_wells_real_reg2.sum())
print(f"Unidades (mil barriles): {best2_wells_real_reg2.sum():.2f}")
print(f"Ganancias: {revenue_reg2:.2f}")

Unidades (mil barriles): 27589.08
Ganancias: 24150866.97


In [39]:
# Calcular ganancias de la región 3 con los datos reales
revenue_reg3 = revenue(best2_wells_real_reg3.sum())
print(f"Unidades (mil barriles): {best2_wells_real_reg3.sum():.2f}")
print(f"Ganancias: {revenue_reg3:.2f}")

Unidades (mil barriles): 28245.22
Ganancias: 27103499.64


- La región en la que propongo invertir en 200 pozos es la región 1 con mayor cantidad de unidades de barriles de petroleo y en consecuencia mejores ganancias.

# Paso 5. Calcular riesgos y ganancias para cada región

## 5.1 bootstrapping

### Región 1

In [40]:
revenues = []

state = np.random.RandomState(12345)

for i in range(1000):
    subsample = best2_wells_real_reg1.sample(n=500, replace=True, random_state=state)
    subsample = pd.Series(subsample).sort_values(ascending=False).head(200)
    revenues.append(revenue(subsample.sum()))

print(f"Tamaño: {len(revenues)}\n")
print(revenues[:20])

Tamaño: 1000

[54214439.03278029, 52760333.06973806, 52350377.275027424, 54345368.055711985, 52756664.476940185, 52915622.01357308, 53299997.50866029, 52715084.426392466, 52123159.69999853, 52864642.240559846, 53699246.92350283, 53457776.40918666, 54250736.46475199, 52999023.200228095, 53428784.7056427, 52612337.441782564, 52953176.94760612, 54455429.92809725, 50864615.56516519, 52119740.21388519]


### Región 2

In [41]:
revenues_reg2 = []

for i in range(1000):
    subsample = best2_wells_real_reg2.sample(n=500, replace=True, random_state=state)
    subsample = pd.Series(subsample).sort_values(ascending=False).head(200)
    revenues_reg2.append(revenue(subsample.sum()))

print(f"Tamaño: {len(revenues)}\n")
print(revenues_reg2[:20])

Tamaño: 1000

[24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114, 24150866.966815114]


### Región 3

In [42]:
revenues_reg3 = []

for i in range(1000):
    subsample = best2_wells_real_reg3.sample(n=500, replace=True, random_state=state)
    subsample = pd.Series(subsample).sort_values(ascending=False).head(200)
    revenues_reg3.append(revenue(subsample.sum()))

print(f"Tamaño: {len(revenues)}\n")
print(revenues_reg3[:20])

Tamaño: 1000

[50023738.1103667, 51574093.86577281, 52655633.22074574, 52078273.962524265, 50604117.29198912, 50631704.49534309, 50815503.28240669, 51088608.26420629, 50197294.656286776, 50602921.448341876, 51190839.505128235, 50812503.67915559, 49376090.675530136, 51147604.99134752, 50111608.13103914, 51092875.020940185, 50349952.781212926, 51281665.87520215, 50295730.303043574, 51455311.67435229]


## 5.2 Intervalo de confianza del 95% y el riesgo de pérdida

### Región 1

In [43]:
print(f"Beneficio promedio: {np.mean(revenues):.2f}")

Beneficio promedio: 53173866.53


In [44]:
# Obtener el intervalo de confianza de 95%
rev = pd.Series(revenues)

lower = rev.quantile(0.025)
upper = rev.quantile(0.075)

print("Invervalo de confianza")
print(f"{lower:.2f}")
print(f"{upper:.2f}")

Invervalo de confianza
51543274.61
51971328.48


In [45]:
# Calcular el riesgo de pérdida
loss = sum(revenue < 0 for revenue in revenues)
rev_len = len(rev)
risk_loss = (loss / rev_len) * 100
print(f"Riesgo de pérdida {risk_loss}%")

Riesgo de pérdida 0.0%


### Región 2

In [46]:
print(f"Beneficio promedio: {np.mean(revenues_reg2):.2f}")

Beneficio promedio: 24150866.97


In [47]:
# Obtener el intervalo de confianza de 95%
rev_reg2 = pd.Series(revenues_reg2)

lower_reg2 = rev_reg2.quantile(0.025)
upper_reg2 = rev_reg2.quantile(0.075)

print("Intervalo de confianza")
print(f"{lower_reg2:.2f}")
print(f"{upper_reg2:.2f}")

Intervalo de confianza
24150866.97
24150866.97


In [48]:
# Calcular el riesgo de pérdida
loss_reg2 = sum(revenue < 0 for revenue in revenues_reg2)
risk_loss_reg2 = (loss_reg2 / len(revenues_reg2)) * 100
print(f"Riesgo de pérdida {risk_loss_reg2:.2f}%")

Riesgo de pérdida 0.00%


### Región 3

In [49]:
print(f"Beneficio promedio: {np.mean(revenues_reg3):.2f}")

Beneficio promedio: 51146486.92


In [50]:
# Obtener el intervalo de confianza de 95%
rev_reg3 = pd.Series(revenues_reg3)

lower_reg3 = rev_reg3.quantile(0.025)
upper_reg3 = rev_reg3.quantile(0.075)

print("Intervalo de confianza")
print(f"{lower_reg3:.2f}")
print(f"{upper_reg3:.2f}")

Intervalo de confianza
49117526.07
49605020.02


In [51]:
# Calcular el riesgo de pérdida
loss_reg3 = sum(r < 0 for r in revenues_reg3)
risk_loss_reg3 = (loss_reg3 / len(revenues_reg3)) * 100
print(f"Riesgo de pérdida {risk_loss_reg3:.2f}%")

Riesgo de pérdida 0.00%


## 2.3 Presentación de conclusiones


**La región 1 es mi elección para invertir en los pozos**

- Ya que cumple con el criterio de riesgo menor a 2.5%, siendo la región con el porcentaje de riesgo más bajo de las otras dos regiones.
- Luego de haber filtrado por el criterio anterior, como se solicita en este proyecto, la región 1 tiene la mejor ganancia de las tres regiones, superando a la región 3 por 2 millones en promedio y dejando muy atrás a la región 2. Ganancia: ($51,543,274).
- Finalmente, sí coincide con mi elección en el punto 4.3, ya que fue la región con mayor producción de unidades (miles de barriles) y demostró mejores ganancias.